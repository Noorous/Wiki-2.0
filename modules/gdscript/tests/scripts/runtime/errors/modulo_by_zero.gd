func subtest_int():
	var x: int = 1
	x %= 0
	print(x)

func subtest_vector_2i():
	var v: Vector2i = Vector2i.ONE
	v %= Vector2i.ZERO
	print(v)

func subtest_vector_3i():
	var v: Vector3i = Vector3i.ONE
	v %= Vector3i.ZERO
	print(v)

func subtest_vector_4i():
	var v: Vector4i = Vector4i.ONE
	v %= Vector4i.ZERO
	print(v)

func test():
	subtest_int()
	subtest_vector_2i()
	subtest_vector_3i()
	subtest_vector_4i()
