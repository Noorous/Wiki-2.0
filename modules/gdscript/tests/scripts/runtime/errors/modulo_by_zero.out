GDTEST_RUNTIME_ERROR
>> SCRIPT ERROR at runtime/errors/modulo_by_zero.gd:3 on subtest_int(): Modulo by zero error in operator '%'.
>> SCRIPT ERROR at runtime/errors/modulo_by_zero.gd:8 on subtest_vector_2i(): Modulo by zero error in operator '%'.
>> SCRIPT ERROR at runtime/errors/modulo_by_zero.gd:13 on subtest_vector_3i(): Modulo by zero error in operator '%'.
>> SCRIPT ERROR at runtime/errors/modulo_by_zero.gd:18 on subtest_vector_4i(): Modulo by zero error in operator '%'.
